### PR TITLE
expand DGE documentation within the vulcan scRNAseq workflow

### PR DESCRIPTION
**Pull request by dtm2451**
Adds a help-doc to the DGE-setup step and also adds a, large, devoted section to the main scRNAseq vignette.

Could use some extra eyes for typo-catching and notes on whether the explanations make sense!

(View them from within vulcan.dev.local!  They're just ugly strings within the cwl and metadata.json)

### CODE REVIEW

File: vulcan/lib/server/workflows/scRNAseq.cwl [hunk 1]
```diff
@@ -321,6 +321,7 @@ steps:
   pick_DGE_methods:
     run: ui-queries/diff-exp-sc.cwl
     label: 'Set how DE should be run'
+    doc: 'Sets the cells compared in differential gene expression calculations.  First, (step 1) you will select your question type, and then (step 2, optional) if there are certain cells you wish to exclude, click the subset checkbox and select the cells which you with to KEEP.  Afterwards, the exact needs (step 3, and sometimes also step 4) will depend on your question type selection of step one, but the overall idea is that these steps fill in the specific sets of cells to compare. See the dedicated section of the overall vignette for more detail. Also note: Sections 7 & 8 of the primary inputs contain additional inputs which contribute to the (7) calculation method or (8) post-calculation filtration cutoffs.'
```
Comment (coleshaw) [inline vulcan/lib/server/workflows/scRNAseq.cwl:324]: (nits): I found the following typos:

`For users initerested in adjusting these parameters` should be `For users interested in adjusting these parameters`

`Selecting of which single-cell data` should be `Selecting which single-cell data`?

`performed to idenfiy top markers` should be `performed to identify top markers`

`versus every other cluster in. These` ... seems to be missing something at the end of the sentence? Not sure what the words should be, though...

`Visulaization and Cluster Annotation (optional)` should be `Visualization and Cluster Annotation (optional)`

`the preliminary differntial expression results` should be `the preliminary differential expression results`

`the final signel-cell data object` should be `the final single-cell data object`

`to perforfollow up analysis outside of vulcan.` should be `to perform follow up analysis outside of Vulcan.` ?

(sorry, can't figure out how to do diffs / comments on specific sections of a single line string...)

File: vulcan/lib/server/workflows/scRNAseq.metadata.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
...en' metadata for subsetting and select to keep all clusters except one, and then utilize configuration 1 for performing the calculation. In any case, to perform subsetting, first check the associated checkbox, then select which metadata to use for subsetting, and then select the labels of that metadata that should be *retained*. Afterwards, the scanpy dataset will be subset to only these cells *prior* to differential expression calculations being performed with the user's chosen configuration.\n- Additional input sets that affect DGE calculation or output:\n  - '7 Post UMAP DGE Calculation Inputs': These inputs are passed along to the function underlying literal differential expression calculations, `scanpy.tl.rank_genes_groups()`.\n  - '8 Post Calculation DGE Cutoffs': These inputs establish cutoffs for the 'filtered' version of the DGE output, which is generated after the 'full' output is constructed.\n\n### Future Feature Addition Plans:\n\n- Additional visualization methods.\n- Back-end updates to improve the user experience by reducing how often analysis steps are re-run.\n\n#### Additional Note: Refer to in-line (?) icons for further information about specific steps/inputs.",
```
Comment (coleshaw) [inline vulcan/lib/server/workflows/scRNAseq.metadata.json:5]: (nits): I found the following typos:

`For users initerested in adjusting these parameters` should be `For users interested in adjusting these parameters`

`Selecting of which single-cell data` should be `Selecting which single-cell data`?

`performed to idenfiy top markers` should be `performed to identify top markers`

`versus every other cluster in. These` ... seems to be missing something at the end of the sentence? Not sure what the words should be, though...

`Visulaization and Cluster Annotation (optional)` should be `Visualization and Cluster Annotation (optional)`

`the preliminary differntial expression results` should be `the preliminary differential expression results`

`the final signel-cell data object` should be `the final single-cell data object`

`to perforfollow up analysis outside of vulcan.` should be `to perform follow up analysis outside of Vulcan.` ?

(sorry, can't figure out how to do diffs / comments on specific sections of a single line string...)

File: vulcan/lib/server/workflows/scRNAseq.cwl (hunk 1 above)
Comment (coleshaw) [inline vulcan/lib/server/workflows/scRNAseq.cwl:324]: (more nits):

`select the cells which you with to KEEP.` should be `select the cells which you wish to KEEP.`

File: vulcan/lib/server/workflows/scRNAseq.metadata.json (hunk 2 above)
Comment (dtm2451) [reply vulcan/lib/server/workflows/scRNAseq.metadata.json]: Thanks!  fixed all these

File: vulcan/lib/server/workflows/scRNAseq.cwl (hunk 1 above)
Comment (dtm2451) [reply vulcan/lib/server/workflows/scRNAseq.cwl]: :+1: fixed

**Comment by dtm2451** [comment]
@bsamad @graft any notes today before I push tomorrow?
Here are the current renderings
## In-line
in-line_DGE_doc
## Within the main vignette
full-vignette_DGE_doc

**Pull request merged**